src/reducers/versions.tsx

```diff
@@ -107,6 +107,15 @@ type PartialExternalVersion = {
   version: string;
 };
 
+export const changeTypes: Record<string, ChangeType> = Object.freeze({
```

Comment:
I did not use `enum` here, because I would like to restrict the value type to `ChangeType` and did not a way to do it in `enum`.

Reply:
I re-considered using `enum` here (in `version.tsx`) and it may be OK without restricting the type of value in defining `enum`, since anyway TS will make sure correct value (ChangeType) is passed to `react-diff-view`. Here is the patch on top of 548486c592237c0e497f4caa56859c027e560bf5
<details> <summary> patch</summary>

```diff 
diff --git a/src/components/DiffView/index.tsx b/src/components/DiffView/index.tsx
index 4691031..6a096d6 100644
--- a/src/components/DiffView/index.tsx
+++ b/src/components/DiffView/index.tsx
@@ -2,6 +2,7 @@ import queryString from 'query-string';
 import * as React from 'react';
 import {
   ChangeInfo,
+  ChangeType,
   Decoration,
   Diff,
   DiffInfo,
@@ -84,7 +85,8 @@ export const trimHunks = ({
 };
 
 export const changeCanBeCommentedUpon = (change: ChangeInfo) => {
-  return [changeTypes.insert, changeTypes.normal].includes(change.type);
+  const types: ChangeType[] = [changeTypes.insert, changeTypes.normal];
+  return types.includes(change.type);
 };
 
 export type PublicProps = {
diff --git a/src/reducers/versions.tsx b/src/reducers/versions.tsx
index 652e21b..eda2780 100644
--- a/src/reducers/versions.tsx
+++ b/src/reducers/versions.tsx
@@ -107,14 +107,14 @@ type PartialExternalVersion = {
   version: string;
 };
 
-export const changeTypes: Record<string, ChangeType> = Object.freeze({
-  delete: 'delete',
-  deleteEofnl: 'delete-eofnl',
-  insert: 'insert',
-  insertEofnl: 'insert-eofnl',
-  normal: 'normal',
-  normalEofnl: 'normal-eofnl',
-});
+export enum changeTypes {
+  delete = 'delete',
+  deleteEofnl = 'delete-eofnl',
+  insert = 'insert',
+  insertEofnl = 'insert-eofnl',
+  normal = 'normal',
+  normalEofnl = 'normal-eofnl',
+}
 
 export type ExternalChange = {
   content: string;
```
</details>

Reply:
In this case these `changeTypes` are not connected in any way with `ChangeType` in `index.d.ts`, correct? This feels even hackier to me than the previous iteration, so I'm not sure it's an improvement. See also my response to your other comment.

Reply:
> In this case these changeTypes are not connected in any way with ChangeType in index.d.ts, correct?

Yes, unfortunately...